pandas/_libs/parsers.pyx

```diff
@@ -2227,9 +2227,10 @@ def _concatenate_chunks(list chunks):
     for name in names:
         arrs = [chunk.pop(name) for chunk in chunks]
         # Check each arr for consistent types.
-        dtypes = set(a.dtype for a in arrs)
-        if len(dtypes) > 1:
-            common_type = np.find_common_type(dtypes, [])
+        dtypes = set([a.dtype for a in arrs])
```

Comment:
Can you rewrite this as a set comprehension, i.e. `{a.dtype for a in arrs}`.  Seems to be preferred based on #18383.